docs/architecture/incremental.md

```diff
@@ -0,0 +1,22 @@
+# Configuration: Streams & Incremental Syncs
+
+## AirbyteStream
```

Comment:
Just an idea. It is a bit confusing to use AirbyteCatalog for something that come from the source and something that is fed back to the source. 

What about we do (while reusing types):
- sources produce an `AirbyteCatalog` which only contains a list of stream
- sources consume a `ConfiguredAirbyteCatalog` which contains the list of `ConfiguredAirbyteStreams`? 

This way it is clearer that one is R/O while the other one got some configuration applied to it. ConfiguredAirbyteCatalog should ideally be backward compatible with AirbyteCatalog (not sure how long that can hold) which would mean: sync everything with full refresh.

WDYT?

Reply:
I think this is a bigger change. We can try it though.

Reply:
i will try this tomorrow. think it's bigger than i have brain to do now.

Reply:
i think this just changes the project from something that could be done last night to something that'll take a couple days. i think it's better. i had considered doing this but thought it was worth the time since this is the chokepoint for all work on incremental. we can folks focus on other stuff while we massage this. will try this approach today.

Reply:
implemented this.